File: 1281.cpp
#include <cstdio>
#include <map>
#include <string>
#include <iostream>

using namespace std;

int main()
{
	map<string , float> dic ; 
	int n , m , p  , quantidade;
	string fruta;
	float preco;
	float total;
	
	cin >> n;
	
	for(int i = 0 ;  i < n ; i ++)
	{

		cin >> m;
		for (int j = 0; j < m; j++)
		{
			cin >> fruta >> preco ;
			dic[fruta] = preco ; 
		}
		cin >> p ;
		
		total = 0 ;
		for (int j = 0; j < p; j++)
		{
			cin >> fruta >> quantidade ;
			total += dic[fruta]*quantidade ;
		}
		
		printf("R$ %.2f\n",total);
		dic = map<string , float>();		
	}
	
	return 0 ; 
}

File: 1281.py
# -*- coding: utf-8 -*-

dic = {}
n = int(input())

for i in range(n):
	m = int(input())
	dic = {}
	for j in range(m):
		entrada = input().split()
		fruta = entrada[0]
		preco = float(entrada[1])
		dic[fruta] = preco
	p = int(input())
	
	total = 0
	for i in range(p):
		entrada = input().split()
		fruta = entrada[0]
		quantidade = int(entrada[1])
		total+= dic[fruta]*quantidade
	total = "%.2f" % (total)
	print("R$ " + total)
